spanner_dbapi/connection.py

```diff
@@ -5,131 +5,98 @@
-        self.run_prior_DDL_statements()
-
-        with self._dbhandle.snapshot() as snapshot:
-            res = snapshot.execute_sql(
-                sql, params=params, param_types=param_types
-            )
-            return list(res)
-
-    def get_table_column_schema(self, table_name):
-        rows = self.run_sql_in_snapshot(
-            """SELECT
-                COLUMN_NAME, IS_NULLABLE, SPANNER_TYPE
-            FROM
-                INFORMATION_SCHEMA.COLUMNS
-            WHERE
-                TABLE_SCHEMA = ''
-            AND
-                TABLE_NAME = @table_name""",
-            params={"table_name": table_name},
-            param_types={"table_name": spanner.param_types.STRING},
-        )
+class Connection(object):
+    """This is a wrap-around object for the existing `Database` and the
+    corresponding `Instance` objects.
 
-        column_details = {}
-        for column_name, is_nullable, spanner_type in rows:
-            column_details[column_name] = ColumnDetails(
-                null_ok=is_nullable == "YES", spanner_type=spanner_type
-            )
-        return column_details
+    :type database: :class:`~google.cloud.spanner_v1.database.Database`
+    :param database: Corresponding Database.
 
-    def close(self):
-        self.rollback()
-        self.__dbhandle = None
-        self._closed = True
+    :type instance: :class:`~google.cloud.spanner_v1.instance.Instance`
+    :param instance: The instance that owns the database.
+    """
 
-    def commit(self):
-        self.__raise_if_already_closed()
+    def __init__(self, database, instance):
+        self.database = database
+        self.instance = instance
+        self.autocommit = True
+        self.read_only = False
+        self.transaction_mode = (
+            TransactionModes.READ_ONLY
+            if self.read_only
+            else TransactionModes.READ_WRITE
+        )
+        self.autocommit_dml_mode = AutocommitDMLModes.TRANSACTIONAL
+        self.timeout_secs = 0
+        self.read_timestamp = None
+        self.commit_timestamp = None
+        self._is_closed = False
+        self._inside_transaction = not self.autocommit
+        self._transaction_started = False
+        self.read_only_staleness = {}
+
+    @property
+    def is_closed(self):
+        return self._is_closed
+
+    @property
+    def inside_transaction(self):
+        return self._inside_transaction
+
+    @property
+    def transaction_started(self):
+        return self._transaction_started
+
+    def _change_transaction_started(self, val: bool):
+        if self._inside_transaction:
+            self._transaction_started = val
 
-        self.run_prior_DDL_statements()
+    def close(self):
+        """Closing database connection"""
+        self._is_closed = True
+
+    @_connection_closed_check
+    def cursor(self):
+        """Returns cursor for current database"""
+        return Cursor(self)
 
+    @_connection_closed_check
```

Comment:
Since this method does not do much and is present merely for standard compatibility, this decorator is redundant and can be removed.